### ISSUE TITLE
U32 values

### ISSUE THREAD

**Issue by andresp** [chrishrb/hoval-gateway#12]
Thanks for putting this project together! I'm in the process of getting data from two Hoval UltraSource B Comfort C heat pumps and I am particularly interested in the energy consumption/production datapoints of the systems. These are found in FunctionGroup 10, FunctionNum 1 and DataPoint Ids 23009 and 29050.

Both these datapoints are 32bit unsigned ints per the specification from Hoval. Interestingly, I can't seem to observe any response messages (42) for any of these datapoints. A can dump of the bus only shows messages that seem to be significantly formatted in a different fashion than the usual response messages. Additionally, the values would have to be sent as two messages as well, since this is a 32bit int.

Do you have any pointers or suggestions as to what I should look for?

**Comment by chrishrb** [comment]
can you maybe post a part of the can dump? maybe I find something useful 😄

**Comment by andresp** [comment]
Sure thing: this is wireshark capture from Heat Pump unit 2 while enumerating the interesting values through a display unit: DP 23009 is are byte values 0x59,0xE1 . There are a few messages that contain these hex values, but these don't look like any of the usual responses.

[wez2.pcapng.zip](https://github.com/chrishrb/hoval-gateway/files/13846773/wez2.pcapng.zip)

**Comment by chrishrb** [comment]
hm can you maybe create a dump with the `candump` tool? (https://sgframework.readthedocs.io/en/latest/cantutorial.html) i'm not really familiar with wireshark^^

**Comment by andresp** [comment]
It may be a few days until I get to capture another dump, so here's a few excerpts from the file above:

```sh
❯ tshark -r wez2.pcapng -V |grep "59 e1"
0000  3d 59 e0 b8 59 e1 b8 59                           =Y..Y..Y
0000  5e 59 e0 b8 59 e1 b8 59                           ^Y..Y..Y
0000  68 59 e0 b8 59 e1 b8 59                           hY..Y..Y
0000  01 40 0a 01 59 e1                                 .@..Y.
0000  11 a8 42 0a 01 59 e1 00                           ..B..Y..
0000  19 a9 56 0a 01 59 e1 f0                           ..V..Y..
```

I'm also attaching a decoded text dump of the file above:

`tshark -r wez2.pcapng -V > wez2.txt`

[wez2.txt](https://github.com/chrishrb/hoval-gateway/files/13847619/wez2.txt)

**Comment by tweak19** [comment]
Any future investigation in this topic? I am facing the same issue.
Need to get the information about the power consumation out of my heating pump.

thx!

**Comment by tweak19** [comment]
Any Updates on that?
I would need the values 29050 & 29051

**Comment by chrishrb** [comment]
yes, there are good news. im currently working on a go rewrite (and im almost at the point where I can test with my HV). 32 bit values are not correctly implemented in the python version. I have to implement this part of the spec:

<img width="468" height="172" alt="Image" src="https://github.com/user-attachments/assets/bb896b32-3f66-41f4-bfaf-46358cd846ba" />

**Comment by chrishrb** [comment]
can you maybe try to create another candump with this tool: https://sgframework.readthedocs.io/en/latest/cantutorial.html
Every new device is helpful 👍 thanks!

**Comment by chrishrb** [comment]
@tweak19 or @andresp do you have any response examples of u32 values? the one that you provided are not responses and in the data there are no datapoints found..

**Comment by tweak19** [comment]
Hi chris,
those datapoint should be for power of the heating pump.
I will capture today a file with candump.

**Comment by tweak19** [comment]
Hi,
now the files.
But you are right, i didn't see any value with 29051...

[can_log_01.txt](https://github.com/user-attachments/files/21967002/can_log_01.txt)

And with current polling the value 29051 i get those feedback from candump:

  20.965105 1  1F400FFFx       Rx   d 8 11 E0 42 0A 01 71 7B 00
  20.982855 1  1F400FFFx       Rx   d 8 19 E1 56 0A 01 71 7B F0
  50.963230 1  1F400FFFx       Rx   d 8 11 80 42 0A 01 71 7B 00
  50.980982 1  1F400FFFx       Rx   d 8 19 81 56 0A 01 71 7B F0
  80.967234 1  1F400FFFx       Rx   d 8 11 D6 42 0A 01 71 7B 00
  80.984981 1  1F400FFFx       Rx   d 8 19 D7 56 0A 01 71 7B F0
 110.970102 1  1F400FFFx       Rx   d 8 11 93 42 0A 01 71 7B 00
 110.989104 1  1F400FFFx       Rx   d 8 19 94 56 0A 01 71 7B F0
 129.192643 1  1F400FFFx       Rx   d 8 11 B9 42 0A 01 71 7B 00

**Comment by chrishrb** [comment]
hm.. do you have a hoval bm (https://www.hoval.de/de_DE/Heiztechnik/Regelung/Systemregelung-TopTronic-E/TopTronic-E-Bedienung/TopTronic-E-RaumbedienModule/p/B_operation-modules-tte) to somehow trigger these requests from there?
i try to trigger u32/s32 requests on my hv but i don't know if this works because as I said I never needed u32/s32 values so I didn't investigate into this..
because its also possible that these values are not triggered by sending a message with 0x40 but instead with something different..

**Comment by chrishrb** [comment]
so yesterday I tried getting responses from my Hoval HV from the BM. I saw the GET requests from the BM but no response from the HV for the U32/S32 values. That makes no sense..

**Comment by tweak19** [comment]
Hi,
yes i got a bm.
but on the bm i see those values for the power, so i would expect they must be anywhere on the canbus.

**Comment by chrishrb** [comment]
yea same for me. i also see the valueson the BM and got some candumps of the bus. But I don‘t see the values on the Bus.. only the request

**Comment by tweak19** [comment]
Thats really interessting.
Do you have any mechanism to find the correct value?
The Problem is, that the heating-pump has a modulation, so you didn't have a static value for the power consumation.
